python/cudf/cudf/utils/ioutils.py

```diff
@@ -1720,6 +1748,30 @@ def get_reader_filepath_or_buffer(
 
     path_or_data = stringify_pathlike(path_or_data)
 
+    if use_python_file_object == no_default:
+        use_python_file_object = False
+    elif use_python_file_object is not None:
+        warnings.warn(
+            "The 'use_python_file_object' keyword is deprecated and "
+            "will be removed in a future version.",
+            FutureWarning,
+        )
+    else:
+        # Preserve the readers (e.g. read_csv) default of True
+        # if no use_python_file_object option is specified by the user
+        # for now (note: this is different from the default for this
+        # function of False)
+        # TODO: when non-pyarrow file reading perf is good enough
+        # we can default this to False
+        use_python_file_object = True
```

Comment:
Just to clarify: This essentially means that this PR is not changing anything in terms of performance, it is just adding a deprecation warning whenever the user sets `use_python_file_object` to anything other than `None` in top-level APIs like `read_parquet`. Am I understanding correctly?

Reply:
Yep, the default will still be ``use_python_file_object=True`` for now if nothing is passed by the user.
(They won't get a deprecation warning though unless they are explicitly passing it in)

Reply:
Okay - That sounds good to me.

I will start revising https://github.com/rapidsai/cudf/pull/16166 to depend on this soon.